library/src/test/java/org/apache/apex/malhar/lib/fs/GenericFileOutputOperatorTest.java

```diff
@@ -121,6 +121,27 @@ public void testTupleCountFinalize() throws IOException
     }
   }
 
+  @Test
+  public void testRotationWithNoData() throws InterruptedException
+  {
+    GenericFileOutputOperator writer = new GenericFileOutputOperator();
+    File dir = new File(testMeta.getDir());
+    writer.setFilePath(testMeta.getDir());
+    writer.setOutputFileName("output.txt");
+    writer.setMaxIdleWindows(5);
+    writer.setAlwaysWriteToTmp(true);
+    writer.setup(testMeta.testOperatorContext);
+
+    for (int i = 0; i < 30; ++i) {
+      writer.beginWindow(i);
+      Thread.sleep(100L);
```

Comment:
Can we reduce this sleep? Or even get rid of it if that makes sense for the test?
If the check is to see no data is sent in the windows, then that should work?